staging/src/k8s.io/api/admissionregistration/v1alpha1/types.go

```diff
@@ -201,6 +201,12 @@ type ValidatingAdmissionPolicySpec struct {
 	// +listMapKey=name
 	// +optional
 	MatchConditions []MatchCondition `json:"matchConditions,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,6,rep,name=matchConditions"`
+
+	// Variables contain definitions of variables that can be used in composition of other expressions.
+	// The variables defined here will be available under `variables` in other expressions.
+	// +listType=atomic
```

Comment:
Why not `map` with a key on `name`?

Reply:
If it is a map, will the order change during applying? The order is very important. I believe it is safer to make it atomic to avoid any surprise with apply.

Reply:
Actually, no. Merging list will preserve the order. I personally fix the SSA ordering issue. I think making it a map makes more sense.